components/registration-framework/org.wso2.carbon.identity.user.registration.mgt/src/main/java/org/wso2/carbon/identity/user/registration/mgt/utils/GraphBuilder.java

```diff
@@ -112,17 +111,13 @@ public static RegistrationGraphConfig convert(RegistrationFlowDTO flowDTO)
         return flowConfig;
     }
 
-    private static String handleEndNode(Map<String, NodeConfig> nodeMap) {
-
-        NodeConfig userOnboardingNode = createUserOnboardingNode();
-        nodeMap.put(userOnboardingNode.getId(), userOnboardingNode);
-        return userOnboardingNode.getId();
-    }
-
     private static NodeConfig processRedirectionStep(StepDTO step, Map<String, NodeConfig> nodeMap,
-                                                     List<NodeEdge> nodeMappings, String endNodeId)
+                                                     List<NodeEdge> nodeMappings)
             throws RegistrationFrameworkException {
 
+        if (step.getData() == null) {
+            throw handleClientException(Constants.ErrorMessages.ERROR_CODE_STEP_DATA_NOT_FOUND, step.getId());
```

Comment:
Prefer to return a server error, as client can't do anything in this context based on the error

Reply:
This is executed during the configuration time of the flow. Therefore the misconfiguration would be an client exception